### ISSUE TITLE
[solved] use_frameworks! and abstract target with dependencies of a main target generate duplicate pod error

### ISSUE THREAD

**Issue by aledalgrande** [CocoaPods/CocoaPods#6540]
* [x] I've read and understood the [*CONTRIBUTING guidelines and have done my best effort to follow](https://github.com/CocoaPods/CocoaPods/blob/master/CONTRIBUTING.md).

# Report

## What did you do?

- added `CocoaLumberjack` to a working Podfile
- ran `pod install`

The Podfile is like this:

```ruby
inhibit_all_warnings!
use_frameworks!

target 'MyApp' do
  # Pods for MyApp
  pod 'bla1'
  pod 'bla2'
  pod 'CocoaLumberjack', '~> 3.1.0'

#  target 'MyAppTests'
end

abstract_target 'My Framework' do
  pod 'bla1'
  pod 'CocoaLumberjack', '~> 3.1.0'

  target 'myFramework1'
  target 'myFramework2'
  target 'myFramework2Tests'
  target 'myFramework3'
  target 'myFramework3Tests'
  target 'myFramework4'
  target 'myFramework5'
  target 'myFramework6'
end
```

In my Xcode project, `MyApp` depends on all the `myFrameworkX` targets (6 in total, all dynamic ObjC frameworks).

## What did you expect to happen?

CocoaLumberjack gets installed successfully and linked to all targets. This already worked with pod `bla1` (which is a dynamic framework).

## What happened instead?

Installation fails with:

> [!] The 'Pods-MyApp' target has frameworks with conflicting names: cocoalumberjack.

From what I know, this should only happen if I try to use two versions of the same pod in interdependent projects. The curious thing is if I remove `CocoaLumberjack` from the `MyApp` target, I still get the error.

I went to print out the target pods for the aggregate target `MyApp` in `install.rb` with a simple `p frameworks` and it actually prints out a list that includes `CocoaLumberjack` 6 times. It would be fine, if that list wasn't then used to check if there are duplicate pods. This is fishy.

Am I not supposed to use a target like this, if there is a dependency between Xcode targets?

All of this was temporarily fixed by removing `use_frameworks!` from the Podfile.

### Stack

```
   CocoaPods : 1.2.0
        Ruby : ruby 2.4.0p0 (2016-12-24 revision 57164) [x86_64-darwin16]
    RubyGems : 2.6.8
        Host : Mac OS X 10.12.3 (16D32)
       Xcode : 8.2.1 (8C1002)
         Git : git version 2.8.3
Ruby lib dir : /usr/local/var/rbenv/versions/2.4.0/lib
Repositories : master - https://github.com/CocoaPods/Specs.git @ 05d8445f92b536799d6d5e78bc86ebe34e478ee5
```

### Installation Source

```
Executable Path: /usr/local/var/rbenv/versions/2.4.0/bin/pod
```

### Plugins

```
cocoapods-deintegrate : 1.0.1
cocoapods-plugins     : 1.0.0
cocoapods-search      : 1.0.0
cocoapods-stats       : 1.0.0
cocoapods-trunk       : 1.1.2
cocoapods-try         : 1.1.0
```

**Comment by aledalgrande** [comment]
## Update

TLDR: always use platform version in the Podfile

I noticed installing without the `frameworks` flag enabled created these two folders:
```
Pods/Target Support Files/CocoaLumberjack-iOS10.0/
Pods/Target Support Files/CocoaLumberjack-iOS10.2/
```
which means different versions of the pod were actually installed. Still don't understand how that's possible, as no other pod had a dependency on it, but probably a podspec has 10.0 requirement, while my Xcode targets have 10.2.

So I added this to the `platform` parameter to the `Podfile`, while also using `use_frameworks!` and `pod install` was **successful** and I only had one version:
```
Pods/Target Support Files/CocoaLumberjack/
```

**Comment by benasher44** [comment]
@aledalgrande glad you got it worked out, and we appreciate you updating this issue with the solution! Have a great day!
